grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -114,10 +114,11 @@ class StockMovementService {
         Boolean statusOnly =
                 jsonObject.containsKey("statusOnly") ? jsonObject.getBoolean("statusOnly") : false
 
+        Comment comment = null
```

Comment:
I don't mind this, but I think it would be maybe better to make it as a default value of the method?
I leave the decision for you

Reply:
I could add default value to a comment but it wouldn't change much.
It might be difficult to see on github view why I am defining a null comment at the begging but in short code looks something like this

```
Comment comment = null

switch(status){
  case CREATED:
    break
  case REQUESTED:
    break
  ....
  case REJECTED:
    comment = new Comment(parameters)
    break
   ...
}

updateRequisitionStatus(stockMovement.id, requisitionStatus, comment)

```

`updateRequisitionStatus` is called at the end for all status transitions, and I only need to pass the comment on REJECT.
So even if I instantiate the comment on REJECT I still need to access it in the outer scope so that's why I defined it at the top.

Reply:
As I said, I don't mind it, it was only a suggestion to let you decide. I'm fine with it as it is currently.